server/sonar-server/src/main/java/org/sonar/server/permission/ws/WsProjectRef.java

```diff
@@ -21,49 +21,54 @@
 package org.sonar.server.permission.ws;
 
 import com.google.common.base.Optional;
-import javax.annotation.Nullable;
+import javax.annotation.CheckForNull;
 import org.sonar.api.server.ws.Request;
 
+import static org.sonar.server.permission.ws.Parameters.PARAM_PROJECT_ID;
 import static org.sonar.server.permission.ws.Parameters.PARAM_PROJECT_KEY;
-import static org.sonar.server.permission.ws.Parameters.PARAM_PROJECT_UUID;
 import static org.sonar.server.ws.WsUtils.checkRequest;
 
 /**
  * Project identifiers from a WS request. Guaranties the project id and project key are not provided at the same time.
  */
-class WsProject {
+class WsProjectRef {
   private final String uuid;
   private final String key;
 
-  private WsProject(String uuid, String key) {
+  private WsProjectRef(Request wsRequest) {
+    String uuid = wsRequest.param(PARAM_PROJECT_ID);
+    String key = wsRequest.param(PARAM_PROJECT_KEY);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "key" which hides the field declared at line 36. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)